platform/docs/docs/migration-guide.md

```diff
@@ -0,0 +1,492 @@
+---
+sidebar_position: 10
+sidebar_label: Migration Guide
+---
+
+# Migration Guide
+
+On this page, we will provide a guide to migrating from OHIF v2 to v3. Please note
+that this document is a work in progress and will be updated as we move forward.
+
+
+## Summary of Changes
+
+OHIF v3 is a complete re-architecture of the OHIF v2 to make it more modular and
+easier to maintain. The main differences are:
+
+- Extensions don't inject their modules into the viewer, they will make them available
+  to be used by the viewer.
+- To use the modules provided by the extensions, you need to write a [Modes](./platform/modes/index.md). Modes
+are configuration objects that will be used by the viewer to load the modules.
+- App configuration structure is different, mainly the `servers` is renamed to `dataSources`.
+- The viewer UI is completely re-written in tailwindcss for better maintainability.
+- cornerstone-core and cornerstone-tools are deprecated and OHIF v3 is using the new Cornerstone3D rendering library and tools
+- A new CLI tool to help you create extensions and modes.
+
+New significant additions that might be useful for you that weren't available in OHIF v2:
+- [OHIF CLI](./development/ohif-cli.md)
+- [New Rendering Engine and Toolings](https://www.cornerstonejs.org/)
+- [Modes](./platform/modes/index.md)
+- [Mode Gallery](https://ohif.org/modes)
+- [Layouts](./platform/extensions/modules/layout-template.md)
+- [Data Sources](./platform/extensions/modules/data-source.md)
+- [Hanging Protocols](./platform/services/data/HangingProtocolService.md)
+- [URL Params](./configuration/url.md)
+
+
+## Configuration
+
+OHIF v3 has a new configuration structure. The main difference is that the `servers` is renamed to `dataSources`
+and the configuration is now asynchronous.
+
+- `StudyPrefetcher` is not currently supported in OHIF v3.
+- The `servers` object has been replaced with a `dataSources` array containing objects representing different data sources.
+- The cornerstoneExtensionConfig property has been removed, you should use `customizationService` instead.
+- The maxConcurrentMetadataRequests property has been removed in favor of `maxNumRequests`
+- The hotkeys array has been updated with different command names and options, and some keys have been removed.
+- New properties have been added, including `maxNumberOfWebWorkers`, `omitQuotationForMultipartRequest`, `showWarningMessageForCrossOrigin`, `showCPUFallbackMessage`, `showLoadingIndicator`, `strictZSpacingForVolumeViewport`.
+
+## Modes
+
+As mentioned briefly above, modes are configuration objects that will be used by the viewer to load extensions.
```

Comment:
I would love to have a simple analogy for Modes and why people should bother about modes? e.g.
Before foo had to to x, y and z but was limited to a, b, and c. Now with modes foo can do y, z and b.